File: php/features.php
<?php
declare(strict_types=1);

// DragonBall SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class DragonBallFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new DragonBallBaseFeature();
            case "test":
                return new DragonBallTestFeature();
            default:
                return new DragonBallBaseFeature();
        }
    }
}

File: rb/features.rb
# DragonBall SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module DragonBallFeatures
  def self.make_feature(name)
    case name
    when "base"
      DragonBallBaseFeature.new
    when "ratelimit"
      DragonBallRatelimitFeature.new
    when "retry"
      DragonBallRetryFeature.new
    when "test"
      DragonBallTestFeature.new
    when "timeout"
      DragonBallTimeoutFeature.new
    else
      DragonBallBaseFeature.new
    end
  end
end
